OEPCIS-759: fixing issue during the parent id inheritance.

Aravinda93 opened the pull request:
@sboeckelmann 

This PR will fix the inheritance issue associated with AssociationEvent -> AggregationEvent or vice versa. This was happening due to we were resetting the value. Hence, it was not inherited. I have also tested with different event types and seems to be working appropriately.

Kindly request you to review the changes and approve the PR.

Commit:
OEPCIS-759: fixing issue during the parent id inheritance.

Commit:
renaming field and passing appropriate values for better understanding.

sboeckelmann commented on testdata-generator-common/src/main/java/io/openepcis/testdata/generator/model/AbstractEventCreationModel.java:
```diff
@@ -235,16 +235,20 @@ public List<String> referencedParentIdentifier(final List<EventIdentifierTracker
                         t -> parentList.addAll(EventModelUtil.parentIdentifiers(t, epc.getInheritParentCount())));
-        epc.setInheritParentCount(0);
+        //epc.setInheritParentCount(0);
       }
```
please remove line completely, commented code may distract from new issues showing up in the future and result in this line to be added again by mistake.

sboeckelmann commented on testdata-generator-common/src/main/java/io/openepcis/testdata/generator/model/AbstractEventCreationModel.java:
```diff
@@ -284,7 +288,8 @@ public List<String> referencedEpcsIdentifierGenerator(
 
-      if (epc.getParentNodeId() != 0 && epc.getInheritParentCount() != null && epc.getInheritParentCount() > 0) {
+      //Add the identifiers from ParentID Count only if the inheriting event is ObjectEvent or TransformationEvent
+      if (epc.getParentNodeId() != 0 && epc.getInheritParentCount() != null && epc.getInheritParentCount() > 0 && inheritParentIdCountFlag) {
```
`epc.getInheritParentCount() != null` - can't we make null-safe and always have inheritParentCount set to zero by default?
just asking, if == null or != null is an important condition somewhere else, just leave it as it is.

sboeckelmann reviewed (commented):
please check my comments

Commit:
remove commented code and set the default value to 0 for all identifier inheritance.

Aravinda93 commented:
@sboeckelmann

Based on the review, I have removed the commented-out code and also added default values for all types of identifier inheritance. Can you please review these changes again?
